### PR TITLE
[VA-9924] Use Driving Distance for Nearby Vet Centers

### LINKED ISSUE #9924: FE "Other nearby Vet Centers" list on Vet Center location page isn't honoring distance correctly
## Describe the defect
The "Other nearby Vet Centers" list on the Vet Center location page doesn't list locations in order of distance. 

## To Reproduce

Originally reported by Tiffany.Lever@va.gov

1. Navigate to this page: http://preview-prod.vfs.va.gov/preview?nodeId=28177
2. View the "Other nearby Vet Centers list

Note the 4 listed:

Fayetteville, NC Vet Center                115-123 miles from MB VC driving distance  (95.57 ATCF)
Charleston, SC Vet Center                99-104 miles from MB VC driving distance (83.98 ATCF)                            
Jacksonville, NC Vet Center              138-144 miles from MB VC (116 ATCF)
Charlotte Vet Center                         188-191 miles from MB VC (154.95 ATCF)  

Note that Charleston is closer than Fayetteville, yet it is listed 2nd
Also, the Colombia, SC Vet Center is not listed, even though it is closer than the Charlotte VC that is listed: 

Colombia, SC Vet Center       141-154 miles from MB VC  (122.22 ATCF)

## AC / Expected behavior
- [ ] Validate that order listed is linear distance 
- [ ] Verify distances against independents source (google maps?)
- [ ] Front end displays XX max number of facilities 
- [ ] Front end displays locations within 120 miles, sorting nearest to farthest, within max number 
- [ ] Locations must be able to be driven to from origin 

The assumption is that "Other nearby Vet Centers" should honor distance. 
How literally are we defining "nearby"? If this is not expected behavior, perhaps some qualifying text like "Not necessarily listed by distance"

@swirtSJW stated that distance calculated is a straight line (the logic does not compute driving distance)

## Screenshots
image

Google maps distance to Fayetteville VC:
image

Google maps distance to Charleston VC 
image

 ## Additional context
From what point is "nearby' calculated? The lat/long of the origin Vet Center? the center of the zip code? the center of the city? 
I know there is a bounding box vs radius issue, which might be contributing to this error?
I suspected this might be driving distance vs as-the-crow-flies distance, but my quick investigation (see screenshots) doesn't support that.

For reference, here is a VC Location List page in production: https://www.va.gov/west-los-angeles-vet-center/locations/

## Desktop (please complete the following information if relevant, or delete)
Issue is independent of OS/Browser

## Labels
(You can delete this section once it's complete)
- [x] Issue type (red) (defaults to "Defect")
- [ ] CMS subsystem (green)
- [ ] CMS practice area (blue)
- [x] CMS workstream (orange) (not needed for bug tickets)
- [ ] CMS-supported product (black)

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public Websites`
- [X] `⭐️ Facilities`
- [ ] `⭐️ User support`

### PR DESCRIPTION
**Pull request by maxx1128**
## Description

"Other Nearby Vet Centers" don't properly organize and filter based on the distances from the first vet center.

[Mapbox has a matrix endpoint](https://docs.mapbox.com/api/navigation/matrix/#retrieve-a-matrix) that lets you measure the driving distances [from multiple points away from a single one (many to one).](https://docs.mapbox.com/help/getting-started/directions/#matrix-api-requests) This PR adds an additional request to that endpoint so each nearby Vet center has the driving distance away in miles. It then sorts the centers from closest to farthest and removes any further away than the original bounding box radius.

## Original issue(s)

Closes department-of-veterans-affairs/va.gov-cms#9924

## Testing done

Visual, see below

## Screenshots

Taken from [the Myrtle Beach Vet Center page.](http://localhost:3002/preview?nodeId=28177)

### Data with new Endpoint

Screen Shot 2022-08-23 at 9 44 05 AM

### Before and After

Screen Shot 2022-08-23 at 9 48 03 AM

Screen Shot 2022-08-23 at 9 35 37 AM

## Acceptance criteria

- [ ] Vet center distances are measured by driving distance, not "as the crow flies" distance
- [ ] Driving distances are validated by an external source or API (Mapbox)
- [ ] Nearby vet centers are organized by driving distance from closest to furthest.
- [ ] Vet centers further away from the bounding box radius aren't included

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/facility-locator/utils/facilityDistance.js [hunk 1]
```diff
@@ -60,3 +61,26 @@ export function calculateBoundingBox(lat, long, radius) {
...
+  const originalVetCenterCoordinatesParam = originalVetCenterCoordinates.join(
+    ',',
+  );
+  const startingCoordinatesParam = nearbyVetCentersCoordinates
+    .map(loc => loc.join(','))
+    .join(';');
+  let sourcesParam = '';
+
+  // eslint-disable-next-line no-plusplus
+  for (let i = 1; i <= nearbyVetCentersCoordinates.length; i++) {
```
[the remark this answers is not in the document]
Comment (maxx1128) [reply src/applications/facility-locator/utils/facilityDistance.js]: The linter recommended `i += 1` and it didn't throw any errors, so I decided to go with that.

File: src/applications/facility-locator/utils/facilityDistance.js [hunk 2]
```diff
@@ -60,3 +61,25 @@ export function calculateBoundingBox(lat, long, radius) {
...
+) => {
+  const originalVetCenterCoordinatesParam = originalVetCenterCoordinates.join(
+    ',',
+  );
+  const startingCoordinatesParam = nearbyVetCentersCoordinates
+    .map(loc => loc.join(','))
+    .join(';');
+  let sourcesParam = '';
+
+  for (let i = 1; i <= nearbyVetCentersCoordinates.length; i += 1) {
```
Comment (wullaski) [inline src/applications/facility-locator/utils/facilityDistance.js:80]: Looks like you might be able to just use `all` as the value for sources. Then you wouldn't need this for loop. https://docs.mapbox.com/api/navigation/matrix/

**Review by wullaski — CHANGES_REQUESTED** [review]

File: src/applications/facility-locator/utils/facilityDistance.js (hunk 2 above)
Comment (maxx1128) [reply src/applications/facility-locator/utils/facilityDistance.js:80]: I tried that before but it was polluting the data. It included the coordinates for the original vet center, so a distance of `0` was getting mixed into the returned data.

**Review by wullaski — APPROVED** [review]
Looks great! :shipit:

**Review by allisonlu — APPROVED** [review]

File: src/applications/facility-locator/utils/facilityDistance.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (maxx1128) [reply src/applications/facility-locator/utils/facilityDistance.js]: @timwright12 Bump in case you didn't see the fix already.

**Pull request merged**